compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -1457,7 +1441,6 @@ impl<'a, 'll, 'tcx> Builder<'a, 'll, 'tcx> {
             return;
         }
 
-        let ptr = self.pointercast(ptr, self.cx.type_i8p());
```

Comment:
Can pointercast be implemented as nop after these changes?

Reply:
It can still be used to convert between different pointer address spaces, e.g. when casting data pointers to function pointers on harvard architecture targets: https://godbolt.org/z/c4nEb8sTW. (Note that we don't use pointercast for this today; that IR is actually invalid as a result. I'll open a PR to fix it.)

On other targets it'll always do nothing. The LLVM side [has a check](https://github.com/llvm/llvm-project/blob/bbcffb08f0fdc0be8c8cba48410f9cb556ea661d/llvm/include/llvm/IR/IRBuilder.h#L2038-L2039) to avoid creating a new instruction if the pointer is already the correct type.